src/annotator/test/pdf-sidebar-test.js

```diff
@@ -0,0 +1,238 @@
+import $ from 'jquery';
+
+//import events from '../../shared/bridge-events';
+
+import PdfSidebar from '../pdf-sidebar';
+import { $imports } from '../pdf-sidebar';
+
+describe('PdfSidebar', () => {
+  const sandbox = sinon.createSandbox();
+  let CrossFrame;
+  let fakeCrossFrame;
+
+  let fakePDFViewerApplication;
+  let fakePDFContainer;
+  let fakePDFViewerUpdate;
+  const sidebarConfig = { pluginClasses: {} };
+
+  const createPdfSidebar = config => {
+    if (!config) {
+      config = {};
+    }
+    config = Object.assign({}, sidebarConfig, config);
+    const element = document.createElement('div');
+    return new PdfSidebar(element, config);
+  };
+
+  beforeEach(() => {
+    sandbox.stub(PdfSidebar.prototype, '_setupGestures');
+    fakePDFContainer = document.createElement('div');
+    fakePDFViewerUpdate = sinon.stub();
+
+    fakePDFViewerApplication = {
+      appConfig: {
+        appContainer: fakePDFContainer,
+      },
+      pdfViewer: {
+        currentScaleValue: 'auto',
+        update: fakePDFViewerUpdate,
+      },
+    };
+    // See https://github.com/sinonjs/sinon/issues/1537
+    // Can't stub an undefined property in a sandbox
+    window.PDFViewerApplication = fakePDFViewerApplication;
+
+    // From `Sidebar.js` tests
+    fakeCrossFrame = {};
+    fakeCrossFrame.onConnect = sandbox.stub().returns(fakeCrossFrame);
+    fakeCrossFrame.on = sandbox.stub().returns(fakeCrossFrame);
+    fakeCrossFrame.call = sandbox.spy();
+    fakeCrossFrame.destroy = sandbox.stub();
+
+    const fakeBucketBar = {};
+    fakeBucketBar.element = $('<div></div>');
+    fakeBucketBar.destroy = sandbox.stub();
+
+    CrossFrame = sandbox.stub();
+    CrossFrame.returns(fakeCrossFrame);
+
+    const BucketBar = sandbox.stub();
+    BucketBar.returns(fakeBucketBar);
+
+    sidebarConfig.pluginClasses.CrossFrame = CrossFrame;
+    sidebarConfig.pluginClasses.BucketBar = BucketBar;
+  });
+
+  afterEach(() => {
+    delete window.PDFViewerApplication;
+    sandbox.restore();
+    $imports.$restore();
+  });
+
+  // sandbox.stub(window, 'innerWidth').value(1020);
+  context('side-by-side mode configured', () => {
+    it('enables side-by-side mode if config and PDF js are present', () => {
+      const sidebar = createPdfSidebar({
+        enableExperimentalPDFSideBySide: true,
+      });
+      assert.isTrue(sidebar.sideBySide);
+    });
+
+    describe('when window is resized', () => {
+      it('attempts to lay out side-by-side', () => {
+        sandbox.stub(window, 'innerWidth').value(1300);
+        const sidebar = createPdfSidebar({
+          enableExperimentalPDFSideBySide: true,
+        });
+
+        window.dispatchEvent(new Event('resize'));
+
+        // PDFSidebar.fitSideBySide is invoked with no argument, so
+        // `sidebar.lastSidebarLayoutState` is used—its default values
```

Comment:
"is used—its" is confusing to me, could it be re-worded?

Reply:
Will make this clearer!